game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/BattleSteps.java

```diff
@@ -185,22 +214,8 @@
     // remove casualties
     steps.add(REMOVE_CASUALTIES);
     // retreat attacking subs
-    if (attackingUnits.stream().anyMatch(Matches.unitCanEvade())) {
-      if (Properties.getSubmersibleSubs(gameData)) {
-        // TODO: BUG? Should the presence of destroyers be checked?
```

Comment:
I figured out why destroyers weren't being checked and it was correct.  The else after this (where it calls canAttackerRetreatSubs) shouldn't have been checking for destroyers either.  I was able to trigger an error in the UI in that else case.